File: SHIVAMFaceEcomm/Scripts/AngularData/ShoppingCart.js
app.controller("ShoppingCartCtrl", function ($scope, AddToCart, CartToCookieService) {
    $scope.emailvalidation = /^[a-z]+[a-z0-9._]+@[a-z]+\.[a-z.]{2,5}$/;
    $scope.CartItems = [];
    $scope.TotalOfCartItems = 0;
    $scope.years;

    $scope.ccinfo = { type: "fa fa-credit-card" }
    $scope.CustomerInfoList = _CustomerInfo;
    $scope.CustomerAddress = { address1: "", address2: "", city: "", state: "", region: "", type: "", country: "" };

    $scope.CustomerDetails = { firstName: $scope.CustomerInfoList.FirstName, lastName: $scope.CustomerInfoList.LastName, phone: $scope.CustomerInfoList.Phone, email: $scope.CustomerInfoList.Email, cardType: 0, CreditCard: "", CardExpMo: "", CardExpYr: "", userName: "", password: "" };

    $scope.loadyear = function () {
        debugger;
        var year = new Date().getFullYear();
        var range = [];
        range.push(year);

        for (var i = 1; i < 20; i++) {
            range.push(year + i);
        }

        $scope.years = range;
    }


    $scope.loadItemsFromCookie = function () {
        debugger;
        $scope.CartItems = CartToCookieService.getCookieData();
        var total = 0;
        for (var i = 0; i < $scope.CartItems.length; i++) {
            var product = $scope.CartItems[i];
            total += (product.Cost * product.Quantity);
        }
        $scope.TotalOfCartItems = total;


    };

    $scope.updateQuantityOfCartItem = function (Product) {





        CartToCookieService.setCookieData($scope.CartItems);

        var total = 0;

        for (var i = 0; i < $scope.CartItems.length; i++) {
            var product = $scope.CartItems[i];

            if (product.Quantity != undefined && product.Quantity != "") {

                total += (product.Cost * product.Quantity);
            }
        }
        $scope.TotalOfCartItems = total;


    };

    $scope.DeleteCartItem = function (Product) {

        //if (confirm("are you sure you want to delete this item from cart ?") == true) {
        //    for (var i = 0; i < $scope.CartItems.length; i++) {
        //        if ($scope.CartItems[i].ProductId == Product.ProductId) {
        //            $scope.CartItems.splice($.inArray(Product, $scope.CartItems), 1);
        //        }
        //    }
        //    debugger;
        //    CartToCookieService.setCookieData($scope.CartItems);

        //    var total = 0;

        //    for (var i = 0; i < $scope.CartItems.length; i++) {
        //        var product = $scope.CartItems[i];
        //        total += (product.Cost * product.Quantity);
        //    }
        //    $scope.TotalOfCartItems = total;

        //}

        bootbox.confirm("are you sure you want to delete this item from cart ?", function (result) {
            if (result) {
                debugger;
                for (var i = 0; i < $scope.CartItems.length; i++) {
                    if ($scope.CartItems[i].ProductId == Product.ProductId) {
                        $scope.CartItems.splice($.inArray(Product, $scope.CartItems), 1);
                    }
                }
                debugger;
                CartToCookieService.setCookieData($scope.CartItems);

                var total = 0;

                for (var i = 0; i < $scope.CartItems.length; i++) {
                    var product = $scope.CartItems[i];
                    total += (product.Cost * product.Quantity);
                }
                $scope.TotalOfCartItems = total; $scope.$apply();
            }


        });


    };




    $scope.Continue = function () {
        debugger;
        var allDataToSend = { CartItems: $scope.CartItems, CustomerData: $scope.CustomerDetails, CustAddress: $scope.CustomerAddress };
        var saveData = $.ajax({
            type: 'POST',
            contentType: 'application/json; charset=utf-8',
            url: "/ShoppingCart/CompletePurchaseAndCreateSessionForUser",
            data: JSON.stringify(allDataToSend),
            dataType: "json",
            success: function (resultData) {
                console.log(resultData);
                if (resultData.Success == true) {
                    alert("Save Complete");
                    window.location.href = "/ShoppingCart/OrderConfirmation";
                }
                else {
                    alert("Error occurred" + resultData.ex)
                }

            }
            , error: function (jqXHR, exception) {
                var msg = '';

                console.log(jqXHR.responseText);
                if (jqXHR.status === 0) {
                    msg = 'Not connect.\n Verify Network.';
                } else if (jqXHR.status == 404) {
                    msg = 'Requested page not found. [404]';
                } else if (jqXHR.status == 500) {
                    msg = 'Internal Server Error [500].';
                } else if (exception === 'parsererror') {
                    msg = 'Requested JSON parse failed.';
                } else if (exception === 'timeout') {
                    msg = 'Time out error.';
                } else if (exception === 'abort') {
                    msg = 'Ajax request aborted.';
                } else {
                    msg = 'Uncaught Error.\n' + jqXHR.responseText;
                }
                alert(msg);
            },
        });
        //saveData.error(function () { alert("Something went wrong"); });

    };
    $scope.loadItemsFromCookie();




});

File: SHIVAMFaceEcomm/Models/ShoppingCart.cs
﻿ 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SHIVAMFaceEcomm.Models
{

    
    public enum CardType : byte
    {
        CreditCart,
        DebitCart,
        Paypal,
        Cashond
    }
    public class ShoppingCart
    {
        public List<CartItems> CartItems { get; set; }
        public Customerd CustomerData { get; set; }
        public CustomerAdd CustAddress { get; set; }
    }

    public class Customerd
    {
        public string firstName { get; set; }

        public string lastName { get; set; }
        public string phone { get; set; }

        public string email { get; set; }

        public string userName { get; set; }
        public string password { get; set; }

        public CardType cardType { get; set; }

        public string CreditCard { get; set; }
        public string CardExpMo { get; set; }
        public string CardExpYr { get; set; }
    }
    public class CustomerAdd
    {
        public string address1 { get; set; }

        public string address2 { get; set; }
        public string city { get; set; }

        public string state { get; set; }

        public string country { get; set; }
        public string region { get; set; }
        public string Type { get; set; }
    }

    public class CartItems
    {
        public int ProductId { get; set; }

        public string ProductName { get; set; }

        public int Quantity { get; set; }

        public Decimal Cost { get; set; }

        public decimal discount { get; set; }

        public int SupplierID { get; set; }
    }


    public class ReviewModel
    {
        public int ProductId { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public String Review { get; set; }
         
    }

    public class APIResponse
    {
        public int ID { get; set; }
        public bool Success { get; set; }

        public string Ex { get; set; }
    }
}